### ISSUE TITLE
Proposal: secondary, typeless compile option

### ISSUE THREAD

**Issue by brian428** [mgechev/angular2-seed#1098]
I've been using this in my own projects, so I thought I would see if you think this worth considering for a pull request.

Basically, `build.js.dev.ts` can be configured to do an initial fully-typed TS compile, but then, it will do a typeless and cached compile for the next N compiler runs. N is configured as a project config value (`FULL_COMPILE_INTERVAL`). I have this set to 6, but it can be set to anything, including -1 which will always do a full compile. Finally, if a compile error is encountered, it will keep triggering full compile runs to execute until the error is eliminated.

In short, you get a fully typed compile every N runs, with typeless compiles in between. Obviously, this can speed things up quite a bit, while still maintaining tolerable compile-time type checking.

I can paste in or attach the code (it's not that much code), or do a pull request, but first I just thought I'd ask if this is something you think might be useful from a conceptual standpoint.

**Comment by mgechev** [comment]
I really like the idea, especially the part of it of performing type checking each next execution of `build.js.dev` until the error is fixed.

Still, I'd love to get a few more opinions cc @ludohenin @d3viant0ne @NathanWalker @TheDonDope.

In my personal project, I've disabled type-checking completely and I relay on vim and tsu plugin to perform it for me.

**Comment by brian428** [comment]
Right. The main issue I've run into with relying on the IDE to do the type checking is that they generally don't constantly check the entire codebase, but only files that are open. So in my experience, it's not hard to make a change that could affect some other file, and the IDE doesn't flag it because that file isn't open. In that case, the compiler is the last line of defense. I didn't want to give that up, so I settled on the above approach as the best compromise I could come up with. :-)

**Comment by mgechev** [comment]
Yes, this is an issue. We handle this using the production build/running tests. Your suggestion seems like an appropriate addition to me.

**Comment by ludohenin** [comment]
Sounds good to me as well.

**Comment by TheDonDope** [comment]
Since there will still be a 'do-always-compile' option (with `FULL_COMIPLE_INTERVAL = -1`), i do like the idea too. Would that also be the default option, so that we won't break with the current behaviour?

**Comment by brian428** [comment]
Right, my assumption would be to have this off by default, and let people override the value in their project config if they choose to. (I also realized that I probably don't need to make -1 force full compiles, when 0 should work just as well.)
